### ISSUE TITLE
`no-extra-parens` possible documentation issue

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3527]
In the [rule docs](http://eslint.org/docs/rules/no-extra-parens), this is not supposed to give an error, and it doesn't:

```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b())

(/^a$/).test(x);
```

However, if a semicolon is added to the end (or if the second statement is removed), it does give an error.
```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b());

(/^a$/).test(x);
```
or
```js
/*eslint no-extra-parens: 2*/

(function(){} ? a() : b())
```

```
3:1  error  Gratuitous parentheses around expression  no-extra-parens
```

I'm thinking that example should have a semicolon added and moved to the "considered warnings" section, yes?

**Comment by michaelficarra** [comment]
The rule is working properly. The first program isn't doing what you think it is doing. Always use semicolons, even if you think you're familiar with how ASI works.

**Comment by IanVS** [comment]
I don't pretend to know how ASI works, so I definitely stick with semicolons.  :)  I figured the rule was fine, and the docs were wrong due to lack of semicolons.  I just wanted a sanity check due to my aforementioned lack of familiarity with ASI.

**Comment by nzakas** [comment]
You are correct, sir.
